### DISCUSSION TITLE
Integrating ConfuserEx in MsBuild

### DISCUSSION THREAD

**Discussion by NeerajYadav612** [mkaring/ConfuserEx#474] [Q&A]
I am trying to integrate ConfuserEx version 1.6 in MsBuild. It suppose to obfuscate the files automatically but it's not working for me. I have added Confuser.MsBuild v1.6 nuget package to the project.

Can somebody help me if I am missing any step.

**Comment by mkaring** [comment]
You'll find some information in the wiki on how to use that package: https://github.com/mkaring/ConfuserEx/wiki/MSBuild-Integration

Does this help?

**Reply by NeerajYadav612** [reply]
As mentioned in the page I changed the csproj file and create a crproj file and try building in debug configuration. Is there anything else I am missing. 
Also, I have multiple projects in my solution so do I need to create crproj file and made changes in csproj file for each project.

**Reply by mkaring** [reply]
I'm usually using the `Directory.Build.props` to setup the obfuscation for the entire solution and also set a central `*.crproj` file.
Find the documentation on those files [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2022).

After your changes, is it working or not? You have not mentioned that. If it's not working, it would help if you could share your C# project file. If you followed the wiki page, it should work fine.

**Reply by NeerajYadav612** [reply]
What I am doing right now is adding the below code in ClinicService.csproj file.
![image](https://user-images.githubusercontent.com/75608343/162352970-7f3fc82e-72b6-4cb8-bfe2-e2ded56b8525.png)


And creating ClinicService.crproj file and putting it next  to the .csproj file.
![image](https://user-images.githubusercontent.com/75608343/162352883-6072c78d-0b2e-4e61-804d-8b14cf0add09.png)


I am building in debug configuration. it's not working for me. Also giving an error:
Expected "@(_EolNetCoreTargetFrameworkVersions->AnyHaveMetadataValue('Identity', '$(_TargetFrameworkVersionWithoutV)'))" to evaluate to a boolean instead of "", in condition "'@(_EolNetCoreTargetFrameworkVersions->AnyHaveMetadataValue('Identity', '$(_TargetFrameworkVersionWithoutV)'))' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(CheckEolTargetFramework)' == 'true'".	ClinicService
